stdlib/unittest/case.pyi

```diff
@@ -275,7 +281,7 @@ class _AssertLogsContext:
     records: List[logging.LogRecord]
     output: List[str]
     def __init__(self, test_case: TestCase, logger_name: str, level: int) -> None: ...
-    def __enter__(self) -> _AssertLogsContext: ...
+    def __enter__(self) -> _LoggingWatcher: ...
```

Comment:
```suggestion
    def __enter__(self) -> _LoggingWatcher | None: ...
```

Reply:
I'm quite curious on how this can be None?
The watcher is initialized in `_CapturingHandler()`, so either I'm overlooking something or this can't be `None`.

Reply:
It returns `None` if `no_logs` is set:

https://github.com/python/cpython/blob/d968a638fcbf9030c999cfacd4c9bf0656e779c4/Lib/unittest/_log.py#L59-L60

Reply:
thanks, I just checked my local Python 3.8 source code.